ITISFoundation/osparc-simcore
# 🐛 Fixes #1517: project tags are copied when cloned

## What do these changes do?

- 🐛 Project clone was not copying tag references (i.e. original tags are referenced)
![copy-tags](https://user-images.githubusercontent.com/32402063/216169438-912ec629-443d-451a-bc42-91844b949d39.gif)


- ♻️ Refactors ``projects_db`` module of the ``webserver``'s service ``projects`` plugin. This is how it looks now
![image](https://user-images.githubusercontent.com/32402063/216094858-c3da03b0-68a7-459c-abc5-4a7f4723ae9c.png)

## Related issue/s

- Ongoing implementation of #3516 : splits away helpers from many body and cleanup dbapi interface for CRUD calls
- Fixes #1517



## How to test

- create a study
- Add tags
- Clone it
- Tags should get copied
- Change/delete one of the tags in the preferences
- Tag in the cloned study should have changed

## Checklist
